apps/src/block_utils.js

```diff
@@ -1085,13 +1085,13 @@ exports.createJsWrapperBlockCreator = function (
         // automatically based on a block's returnType (e.g. yellow for "Location").
         if (color) {
```

Comment:
could this just be an `else if`? We wouldn't get the warning log in that case (is it useful?), but it would make this logic clearer that style overrides color.

Reply:
It can't be an `else if` because of CDO Blockly. Consider a Minecraft block with both a `color` and `style`. We'd try to set a style (no-op), and then we'd be done.

Reply:
ah gotcha I didn't realize cdo blockly blocks had styles.

Reply:
Minecraft blocks do, as part of the work to get them ready for migration. This code makes it safe for a CDO Blockly block to have both for now (even though we only use the color), but it will automatically start using using the style when we flip the lab.

Reply:
Could we use `getStyleName` instead?

Reply:
(in place of the check on `style`)

Reply:
We could move the `getStyleName` up earlier like this:

```
if (style) {
  // Attempt to set a style. Only works on Google Blockly
  this.setStyle(style);
}
if (!this.getStyleName()) {
  if (color) {
    Blockly.cdoUtils.setHSV(this, ...color);
  } else if (!returnType) {
    // CDO Blockly assigns colors to blocks with an output connection based on return type.
    // See Blockly.Connection.prototype.colorForType
    // Blocks with neither style nor color that do not have a return type can
    // use the default teal color and style.
    Blockly.cdoUtils.setHSV(this, ...DEFAULT_COLOR);
    this.setStyle('default');
  } 
} // else: block has a style 
```

`this.getStyleName()` will tell us if we've already successfully set a style on the block, but only after we've called `setStyle` on it.

Reply:
See this other comment: https://github.com/code-dot-org/code-dot-org/pull/56589#discussion_r1491166313

Reply:
do you think that's cleaner? I'm not sure I see a clear value from a warning log when we ignore a color value on a block, but if that's helpful for curriculum debugging we can keep it as-is!

Reply:
Moved the logic out to a util for each wrapper in hopes that it's clearer:
https://github.com/code-dot-org/code-dot-org/pull/56589/commits/330868f8b4a15d6aa9af0d2f366cacd7fe24fd38